src/App.tsx

```diff
@@ -1,14 +1,71 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
+import { User } from './types/User';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import './types/FilterBy';
+import { ACTIVE, COMPLETED, ALL } from './types/FilterBy';
+import { TodoModal } from './components/TodoModal';
+
+function filterTodos(allTodos: Todo[], query: string, filter: string) {
+  let todos = [...allTodos];
+
+  if (query) {
+    const queryLowerCase = query.toLowerCase();
+
+    todos = todos.filter(todo =>
+      todo.title.toLowerCase().includes(queryLowerCase),
+    );
+  }
+
+  switch (filter) {
+    case ACTIVE:
+      return todos.filter(todo => !todo.completed);
+
+    case COMPLETED:
+      return todos.filter(todo => todo.completed);
+
+    default:
+      return todos;
+  }
+}
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState(ALL);
+  const [showModal, setShowModal] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  const onEyeClicked = (todo: Todo) => {
+    setShowModal(true);
+    getUser(todo.userId).then(userFromApi => {
+      setUser(userFromApi);
+    });
+    setSelectedTodo(todo);
+  };
+
+  const onCloseClicked = () => {
+    setShowModal(false);
+    setUser(null);
+    setSelectedTodo(null);
+  };
+
+  useEffect(() => {
+    getTodos().then(todosFromApi => {
+      setTodos(todosFromApi);
+      setIsLoading(false);
+    });
```

Comment:
Similar to the `getUser` API call, consider adding error handling for the `getTodos` API call. This will help manage any potential issues with fetching data and improve the robustness of your application.